src/main/java/fishman/utils/Parser.java

```diff
@@ -0,0 +1,131 @@
+            case "mark":
+                if (tasks.size() == 0) {
+                    throw new FishmanException.EmptyListException();
+                }
+                int markIndex = Integer.parseInt(inputs[1]) - 1;
+                if (markIndex < 0 || markIndex >= tasks.size()) {
+                    throw new FishmanException.IndexOutOfBoundsException(markIndex + 1);
+                }
+                return new MarkCommand(markIndex, true);
+            case "unmark":
+                if (tasks.size() == 0) {
+                    throw new FishmanException.EmptyListException();
+                }
+                int unmarkIndex = Integer.parseInt(inputs[1]) - 1;
+                if (unmarkIndex < 0 || unmarkIndex >= tasks.size()) {
+                    throw new FishmanException.IndexOutOfBoundsException(unmarkIndex + 1);
+                }
+                return new MarkCommand(unmarkIndex, false);
+            case "todo":
+                return new AddCommand(new ToDo(inputs[1], false));
+            case "deadline":
+                if (!inputs[1].contains("/by")) {
+                    throw new FishmanException.MissingArgumentException("deadline");
+                }
+                String[] deadlineString = inputs[1].split("/by");
+                LocalDateTime deadlineDate = parseDateTime(deadlineString[1].trim());
+                return new AddCommand(new Deadline(deadlineString[0].trim(), false, deadlineDate));
+            case "event":
+                if (!inputs[1].contains("/from") || !inputs[1].contains("/to")) {
+                    throw new FishmanException.MissingArgumentException("event");
+                }
+                String[] eventString = inputs[1].split("/from|/to");
+                LocalDateTime fromDate = parseDateTime(eventString[1].trim());
+                LocalDateTime toDate = parseDateTime(eventString[2].trim());
+                return new AddCommand(new Event(eventString[0].trim(), false, fromDate, toDate));
+            case "delete":
+                int deleteIndex = Integer.parseInt(inputs[1]) - 1;
+                if (deleteIndex < 0 || deleteIndex >= tasks.size()) {
+                    throw new FishmanException.IndexOutOfBoundsException(deleteIndex + 1);
+                }
+                return new DeleteCommand(deleteIndex);
+            case "find":
+                return new FindCommand(inputs[1]);
+            default:
+                throw new FishmanException.InvalidCommandException();
+            }
+        } catch (NumberFormatException e) {
+            throw new FishmanException.NumberFormatException(e.getMessage());
+        }
+    }
+
+    private static LocalDateTime parseDateTime(String dateTimeStr) throws FishmanException {
+        try {
+            return LocalDateTime.parse(dateTimeStr, INPUT_FORMATTER);
+        } catch (DateTimeParseException e) {
+            throw new FishmanException.InvalidDateFormatException(dateTimeStr);
+        }
+    }
```

Comment:
consider adding a Javadoc comment for the private method `parseDateTime` to maintain consistency and provide context for future developers.